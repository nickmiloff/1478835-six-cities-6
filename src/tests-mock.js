export const simpleMock = `test`;

export const cards = {
  changedCard: {
    id: 2,
    isFavorite: true
  },
  start: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: false
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  end: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  endWithRemove: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ]
};

export const auth = {
  api: {
    "id": 1,
    "email": `mail@mail.com`,
    "name": `name`,
    "avatar_url": `avatar.jpeg`,
    "is_pro": true
  },
  adapt: {
    id: 1,
    email: `mail@mail.com`,
    name: `name`,
    avatarUrl: `avatar.jpeg`,
    isPro: true
  }
};

export const user = {
  email: `mail@mail.com`,
  password: 12345678
};

export const cardsFull = {
  api: [
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 2,
      "type": `house`,
      "bedrooms": 2,
      "price": 161,
      "goods": [`Laptop friendly workspace`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "id": 1,
      "is_favorite": false,
      "is_premium": true,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "max_adults": 9
    },
    {
      "city": {
        "name": `Amsterdam`,
        "location": {
          "latitude": 52.37454,
          "longitude": 4.897976,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Loft Studio in the Central Area`,
      "rating": 3.6,
      "type": `apartment`,
      "bedrooms": 3,
      "price": 271,
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 52.388540000000006,
        "longitude": 4.899976,
        "zoom": 16
      },
      "id": 2,
      "is_favorite": false,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    },
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 4.6,
      "type": `house`,
      "bedrooms": 2,
      "price": 865,
      "goods": [`Laptop friendly workspace`, `Breakfast`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 48.83861,
        "longitude": 2.350499,
        "zoom": 16
      },
      "id": 3,
      "is_favorite": true,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    }
  ],
  adapted: [
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 2,
      type: `house`,
      bedrooms: 2,
      price: 161,
      goods: [`Laptop friendly workspace`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16,
      },
      id: 1,
      isFavorite: false,
      isPremium: true,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      maxAdults: 9,
    },
    {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13,
        },
      },
      images: [],
      title: `Loft Studio in the Central Area`,
      rating: 3.6,
      type: `apartment`,
      bedrooms: 3,
      price: 271,
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`,
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 52.388540000000006,
        longitude: 4.899976,
        zoom: 16,
      },
      id: 2,
      isFavorite: false,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 4.6,
      type: `house`,
      bedrooms: 2,
      price: 865,
      goods: [`Laptop friendly workspace`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 48.83861,
        longitude: 2.350499,
        zoom: 16,
      },
      id: 3,
      isFavorite: true,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
  ]
};

export const comments = {
  api: [
    {
      "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      "date": `2019-05-08T14:13:56.569Z`,
      "id": 1,
      "rating": 4,
      "user": {
        "avatar_url": `img/1.png`,
        "id": 4,
        "is_pro": false,
        "name": `Max`
      }
    },
    {
      "comment": `A quiet of Amsterdam.`,
      "date": `2019-06-08T14:13:56.569Z`,
      "id": 2,
      "rating": 3,
      "user": {
        "avatar_url": `img/2.png`,
        "id": 5,
        "is_pro": false,
        "name": `Vadim`
      }
    }
  ],
  adapted: [
    {
      comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: `2019-05-08T14:13:56.569Z`,
      id: 1,
      rating: 4,
      user: {
        avatarUrl: `img/1.png`,
        id: 4,
        isPro: false,
        name: `Max`
      }
    },
    {
      comment: `A quiet of Amsterdam.`,
      date: `2019-06-08T14:13:56.569Z`,
      id: 2,
      rating: 3,
      user: {
        avatarUrl: `img/2.png`,
        id: 5,
        isPro: false,
        name: `Vadim`
      }
    }
  ]
};
