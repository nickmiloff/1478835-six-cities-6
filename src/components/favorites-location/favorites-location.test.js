import React from 'react';
import {render} from '@testing-library/react';
import Test from '../test/test';

import FavoritesLocation from './favorites-location';
import {cardsFull} from '../../tests-mock';

describe(`FavoritesLocation component test`, () => {
  it(`'FavoritesLocation component' should render correctly`, () => {
    const CITY = `Paris`;
    const CARDS = cardsFull.adapted;

    const {container} = render(
        <Test>
          <FavoritesLocation
            city={CITY}
            cards={CARDS} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
