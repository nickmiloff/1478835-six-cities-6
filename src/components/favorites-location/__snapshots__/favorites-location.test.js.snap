// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesLocation component test 'FavoritesLocation component' should render correctly 1`] = `
<div>
  <li
    class="favorites__locations-items"
  >
    <div
      class="favorites__locations locations locations--current"
    >
      <div
        class="locations__item"
      >
        <a
          class="locations__item-link"
          href="#"
        >
          <span>
            Paris
          </span>
        </a>
      </div>
    </div>
    <div
      class="favorites__places"
    >
      <article
        class="favorites__card place-card"
      >
        <div
          class="place-card__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/1"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="110"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
              width="150"
            />
          </a>
        </div>
        <div
          class="favorites__card-info place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                161
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button"
              data-testid="to-bookmarks-button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 40%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/1"
            >
              Tile House
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            house
          </p>
        </div>
      </article>
      <article
        class="favorites__card place-card"
      >
        
        <div
          class="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/2"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="110"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
              width="150"
            />
          </a>
        </div>
        <div
          class="favorites__card-info place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                271
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button"
              data-testid="to-bookmarks-button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 72%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/2"
            >
              Loft Studio in the Central Area
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            apartment
          </p>
        </div>
      </article>
      <article
        class="favorites__card place-card"
      >
        
        <div
          class="favorites__image-wrapper place-card__image-wrapper"
        >
          <a
            href="/offer/3"
          >
            <img
              alt="Place image"
              class="place-card__image"
              height="110"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg"
              width="150"
            />
          </a>
        </div>
        <div
          class="favorites__card-info place-card__info"
        >
          <div
            class="place-card__price-wrapper"
          >
            <div
              class="place-card__price"
            >
              <b
                class="place-card__price-value"
              >
                €
                865
              </b>
              <span
                class="place-card__price-text"
              >
                / night
              </span>
            </div>
            <button
              class="place-card__bookmark-button button place-card__bookmark-button--active"
              data-testid="to-bookmarks-button"
              type="button"
            >
              <svg
                class="place-card__bookmark-icon"
                height="19"
                width="18"
              >
                <use
                  xlink:href="#icon-bookmark"
                />
              </svg>
              <span
                class="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            class="place-card__rating rating"
          >
            <div
              class="place-card__stars rating__stars"
            >
              <span
                style="width: 92%;"
              />
              <span
                class="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <h2
            class="place-card__name"
          >
            <a
              href="/offer/3"
            >
              Tile House
            </a>
          </h2>
          <p
            class="place-card__type"
          >
            house
          </p>
        </div>
      </article>
    </div>
  </li>
</div>
`;
